File: app/src/main/java/com/unero/activity/TampilData.java
package com.unero.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class TampilData extends AppCompatActivity {
    TextView tnim, tnama, tgender, tjurusan, ttglLahir;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tampil_data);

        tnim = findViewById(R.id.txtNIM);
        tnama = findViewById(R.id.txtNama);
        tgender = findViewById(R.id.txtGender);
        tjurusan = findViewById(R.id.txtJurusan);
        ttglLahir = findViewById(R.id.txtTglLahir);

        // Get Data
        Long NIM = getIntent().getLongExtra("NIM", 1L);
        String nama = getIntent().getStringExtra("nama");
        String gender = getIntent().getStringExtra("gender");
        String jurusan = getIntent().getStringExtra("jurusan");
        String tglLahir = getIntent().getStringExtra("tglLahir");

        tnim.setText(String.valueOf(NIM));
        tnama.setText(nama);
        tgender.setText(gender);
        tjurusan.setText(jurusan);
        ttglLahir.setText(tglLahir);

    }
}

File: app/src/main/java/com/unero/activity/TampilData.kt
package com.unero.activity

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class TampilData : AppCompatActivity() {

    private lateinit var nim: TextView
    private lateinit var nama: TextView
    private lateinit var birth: TextView
    private lateinit var gender: TextView
    private lateinit var jurusan: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tampil_data)

        initViews()
        getSet()
    }

    private fun initViews(){
        nim = findViewById(R.id.txtNIM)
        nama = findViewById(R.id.txtNama)
        birth = findViewById(R.id.txtTgl)
        gender = findViewById(R.id.txtGender)
        jurusan = findViewById(R.id.txtJurusan)
    }

    private fun getSet() {
        val mahasiswa = intent.extras?.getParcelable<Mahasiswa>("MHS")
        val vnim: String = mahasiswa!!.nim.toString()
        val vnama: String = mahasiswa.nama
        val vbirth: String = mahasiswa.date
        val vgender: String = mahasiswa.gender
        val vjurusan: String = mahasiswa.jurusan

        nim.text = vnim
        nama.text = vnama
        birth.text = vbirth
        gender.text = vgender
        jurusan.text = vjurusan
    }
}